### PR TITLE
Add facter

### PR DESCRIPTION
**Pull request by womfoo**

### CODE REVIEW

**Comment by jagajaga** [comment]
Please rename commit's name following our guide from `CONTRIBUTING.md`

File: pkgs/tools/system/facter/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  src = fetchurl {
+    url = "https://downloads.puppetlabs.com/facter/${name}.tar.gz";
+    sha256 = "1myf1r827bh3n0snkzwj2jnc0sax9bq6z1vv9gr90rqr73lixvig";
+  };
+
+  buildInputs = [ boost cmake curl libyamlcpp openssl utillinux ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/puppetlabs/facter;
+    description = "Facter, a system inventory tool";
```
Comment (jagajaga) [inline pkgs/tools/system/facter/default.nix]: Please omit pkg name in the description.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -15133,4 +15133,8 @@ let
 
   udevil = callPackage ../applications/misc/udevil {};
 
+  facter = callPackage ../tools/system/facter rec {
+     libyamlcpp = callPackage ../development/libraries/libyaml-cpp { makePIC=true; };
```
Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: Better to override pkg in the `default.nix`

**Comment by womfoo** [comment]
thanks for looking into this @jagajaga. i've just rebased the changes.

**Comment by jagajaga** [comment]
But you forgot to change the commit's name :)

**Comment by womfoo** [comment]
Commit message updated. Thanks @jagajaga

**Comment by lethalman** [comment]
Travis built, thanks.

**Pull request merged**